internal/versions/versions_test.go

```diff
@@ -189,20 +201,93 @@ var _ = Describe("list versions", func() {
 		})
 	})
 
+	Context("GetOCPVersion", func() {
+		var (
+			ocpVersion string
+			err        error
+		)
+
+		BeforeEach(func() {
+			openshiftVersions = &defaultOpenShiftVersions
+			h = NewHandler(logger, mockRelease, workDir, versions, *openshiftVersions, "")
+		})
+
+		It("default", func() {
+			for key := range *openshiftVersions {
+				ocpVersion, err = h.GetOCPVersion(key)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(ocpVersion).Should(Equal(*(*openshiftVersions)[key].DisplayName))
+			}
+		})
+
+		It("unsupported_key", func() {
+			ocpVersion, err = h.GetOCPVersion("unsupported")
+			Expect(err).Should(HaveOccurred())
+			Expect(ocpVersion).Should(BeEmpty())
+		})
+	})
+
 	Context("IsOpenshiftVersionSupported", func() {
 		It("positive", func() {
-			h := NewHandler(logger, mockRelease, versions, *openshiftVersions, "")
+			h := NewHandler(logger, mockRelease, workDir, versions, *openshiftVersions, "")
 
 			for key := range *openshiftVersions {
 				Expect(h.IsOpenshiftVersionSupported(key)).Should(BeTrue())
 			}
 		})
 
 		It("negative", func() {
-			h := NewHandler(logger, mockRelease, versions, *openshiftVersions, "")
+			h := NewHandler(logger, mockRelease, workDir, versions, *openshiftVersions, "")
 			Expect(h.IsOpenshiftVersionSupported("unknown")).Should(BeFalse())
 		})
 	})
+
+	Context("AddOpenshiftVersion", func() {
+		var (
+			pullSecret   = "test_pull_secret"
+			releaseImage = "releaseImage"
+			ocpVersion   = "4.7.0"
+		)
+
+		It("added version successfully", func() {
+			h := NewHandler(logger, mockRelease, workDir, versions, customOpenShiftVersions, "")
+			mockRelease.EXPECT().GetOpenshiftVersion(
+				gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(ocpVersion, nil).AnyTimes()
+
+			version, err := h.AddOpenshiftVersion(releaseImage, pullSecret)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			versionKey, err := h.GetSupportedVersionFormat(ocpVersion)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			versionFromCache := h.openshiftVersions[versionKey]
+			Expect(*version.DisplayName).Should(Equal(ocpVersion))
+			Expect(*version.ReleaseImage).Should(Equal(releaseImage))
+			Expect(version.RhcosImage).Should(Equal(versionFromCache.RhcosImage))
+			Expect(version.RhcosVersion).Should(Equal(versionFromCache.RhcosVersion))
+		})
+
```

Comment:
Add scenario for overriding an existing image

Reply:
done